project/dexter-metrics/src/java/com/samsung/sec/dexter/metrics/util/CdtUtilHelper.java

```diff
@@ -0,0 +1,442 @@
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.samsung.sec.dexter.metrics.util;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.log4j.Logger;
+import org.eclipse.cdt.core.dom.ast.IASTCaseStatement;
+import org.eclipse.cdt.core.dom.ast.IASTCompoundStatement;
+import org.eclipse.cdt.core.dom.ast.IASTDeclSpecifier;
+import org.eclipse.cdt.core.dom.ast.IASTDeclaration;
+import org.eclipse.cdt.core.dom.ast.IASTDeclarator;
+import org.eclipse.cdt.core.dom.ast.IASTDoStatement;
+import org.eclipse.cdt.core.dom.ast.IASTForStatement;
+import org.eclipse.cdt.core.dom.ast.IASTFunctionDeclarator;
+import org.eclipse.cdt.core.dom.ast.IASTFunctionDefinition;
+import org.eclipse.cdt.core.dom.ast.IASTIfStatement;
+import org.eclipse.cdt.core.dom.ast.IASTNode;
+import org.eclipse.cdt.core.dom.ast.IASTSimpleDeclaration;
+import org.eclipse.cdt.core.dom.ast.IASTStatement;
+import org.eclipse.cdt.core.dom.ast.IASTSwitchStatement;
+import org.eclipse.cdt.core.dom.ast.IASTWhileStatement;
+import org.eclipse.cdt.internal.core.dom.parser.cpp.CPPASTCatchHandler;
+import org.eclipse.cdt.internal.core.dom.parser.cpp.CPPASTCompositeTypeSpecifier;
+import org.eclipse.cdt.internal.core.dom.parser.cpp.CPPASTTryBlockStatement;
+
+import com.google.common.io.Files;
+import com.samsung.sec.dexter.core.analyzer.ResultFileConstant;
+
+public class CdtUtilHelper {
+	static int count = 0;
+	static int sloc = 0;
+	static int methodCount = 0;
+	static int classCount = 0;
+	final static Map<String, Integer> mapSourceMatrices = new HashMap<String, Integer>();
+	final static Map<String, Integer> mapFunctionLocMetrices = new HashMap<String, Integer>();
+	public final static String LINE_SEPARATOR = System.getProperty("line.separator");
+	final private static Map<String, String> mapModuleName = new HashMap<String, String>();
+	static Logger logger = Logger.getLogger(CdtUtilHelper.class);
+
+	static Map<String, String> getMapData() {
+		if (mapModuleName.isEmpty()) {
+			mapModuleName.put(ResultFileConstant.CLASS_NAME, "");
+			mapModuleName.put(ResultFileConstant.METHOD_NAME, "");
+		}
+		return mapModuleName;
+	}
+
+	private CdtUtilHelper() {
+
+	}
+
+	public static String getContentsFromFile(final String filePath,
+			final Charset charset) {
+		if (filePath == null || filePath.equals("")) {
+			logger.error("Invalid Parameter : filePath is null or empty");
+			return "";
+		}
+
+		final File file = new File(filePath);
+
+		if (file.exists() == false || file.isDirectory()) {
+
+			return null;
```

Comment:
Please check caller has code for null checking.
The better way is to use empty String object or throw exception.